vk_generator/src/generator/mod.rs

```diff
@@ -718,9 +731,26 @@ impl<'a> GenTypes<'a> {
                                 }
                                 writeln!(structs, "{},", &*ident)
                             }
-                            MutArray(ident, count)    => writeln!(structs, "{}: [{}; {}],", &*f.field_name, &*ident, count),
-                            MutArrayEnum(ident, cons) => writeln!(structs, "{}: [{}; {}],", &*f.field_name, &*ident, &*cons),
-
+                            MutArray(ident, count) => {
+                                let mut fty = &*ident;
+                                if gen_types.config.remove_type_prefix && fty.starts_with("VK_") {
+                                    fty = &fty[3..];
+                                }
+                                writeln!(structs, "{}: [{}; {}],", &*f.field_name, fty, count)
+                            }
+                            MutArrayEnum(ident, cons) => {
+                                let mut fty = &*ident;
+                                let mut cons = &*cons;
+                                if gen_types.config.remove_type_prefix {
+                                    if fty.starts_with("VK_") {
+                                        fty = &fty[3..];
+                                    }
+                                    if cons.starts_with("VK_") {
+                                        cons = &cons[3..];
+                                    }
+                                }
```

Comment:
Again, the `fty.starts_with("VK_")` branch shouldn't ever be executed. Also, constant prefix stripping should be performed in the `add_interface` function, which handles identifier transforms. For stripping it in `ArrayEnum`s, the `add_type_recurse` function has a section that adds the array's length to the `const` list - the stripping should be performed there, before the const gets added.